src/lib.rs

```diff
@@ -412,6 +420,50 @@ fn run_for_context<TContext: Sync>(
     })
 }
 
+pub fn run_for_slice_with_callback(
+    slice: &[u8],
+    args: Args,
+    mut callback: impl FnMut(CaptureInfo) + Sync,
+) -> Result<RunStatus, Error> {
+    let language = args.language.ok_or(Error::LanguageMissingForSlice)?;
```

Comment:
Basically `Args` has different "validation" for this entry point (specifically we need to know which language you want this slice to be treated as)

I started thinking that maybe the slice itself could/should be a field on `Args` and then that would be something that `Args` could "always validate" (ie that if its eg `.slice_to_search` field is present then its `.language` field must also be present)?

But that might be weird because then you'd have to sort of assert that "for this entry point we expect `args.slice_to_search` to be set" and vice-versa?